keras/wrappers/scikit_learn.py

```diff
@@ -149,8 +179,108 @@ class KerasClassifier(BaseWrapper):
     loss : string
         Loss function used by the model during compilation/training.
     """
-    def __init__(self, model, optimizer='adam', loss='categorical_crossentropy', **kwargs):
-        super(KerasClassifier, self).__init__(model, optimizer, loss, **kwargs)
+
+    def __init__(self, 
+                 model, 
+                 optimizer='adam', 
+                 loss='categorical_crossentropy', 
+                 train_batch_size=128, 
+                 test_batch_size=128,
+                 nb_epoch=100, 
+                 shuffle=True, 
+                 show_accuracy=False,
+                 validation_split=0, 
+                 validation_data=None, 
+                 callbacks=None,
+                 verbose=0, 
+                 class_weight=None):
+        super(KerasClassifier, self).__init__(model, 
+                                              optimizer, 
+                                              loss, 
+                                              train_batch_size, 
+                                              test_batch_size,
+                                              nb_epoch, 
+                                              shuffle, 
+                                              show_accuracy,
+                                              validation_split, 
+                                              validation_data, 
+                                              callbacks,
+                                              verbose)
+
+        self.class_weight = class_weight
+
+
+    def __ohe_output(function):
```

Comment:
I would suggest dropping the decorator and instead repeating the code twice. You could abstract it in a helper function if that's really needed. I don't think a decorator is helpful in terms of readability in this case.